### PR TITLE
Conditional reveal

### PR DESCRIPTION
**Pull request by joefitter**
* bumped react-components to 0.3.0 including conditional reveal
* updated exemption schema to include child fields
* added custom validation functionality
* added custom validator to check module selected if reason empty
* split form and fieldset
* create flat model representing exemptions for ui
* post array of exemptions

### CODE REVIEW

File: pages/common/assets/sass/style.scss [hunk 1]
```diff
@@ -295,6 +295,11 @@ dl.inline {
   }
 }
 
+.govuk-checkboxes__item > div {
+  margin-top: $govuk-gutter / 2;
+  margin-left: -24px;
+}
```
Comment (wheelsandcogs) [inline pages/common/assets/sass/style.scss]: Let's fix this in `@ukhomeoffice/frontend-toolkit`?

Comment (joefitter) [reply pages/common/assets/sass/style.scss]: is that where it goes? haha I couldn't find the css in react-components

Comment (joefitter) [reply pages/common/assets/sass/style.scss]: neither, the html is in `react-components`

Comment (joefitter) [reply pages/common/assets/sass/style.scss]: I guess that's what you meant by `ho-react-components`

Comment (joefitter) [reply pages/common/assets/sass/style.scss]: BTW, I think the css should be in the same repo as the react, `react-components` in this case

File: pages/pil/application/views/exemption-data.jsx [hunk 2]
```diff
@@ -14,18 +14,10 @@ class ExemptionData extends Component {
...
-                            {module.module}
-                            {module.exemptionDescription}
-                          </li>
-                        ))
-                      }
-                    </ul>
-                  </dd>
+                  <dd>{ exemption.modules[0].module }</dd>
+
+                  <dt><Snippet>certificate.reason</Snippet></dt>
```
Comment (wheelsandcogs) [inline pages/pil/application/views/exemption-data.jsx]: This content key probably wants renaming when we do the refactor...

File: pages/pil/modules-exempt/index.js [hunk 3]
```diff
@@ -1,29 +1,80 @@
...
 
   app.post('/', (req, res, next) => {
-    const values = omit(req.session.form[req.model.id].values, 'exempt');
-    values.profileId = req.profile;
-    values.modules = values.modules.map(module => ({ module, species: [] }));
-    values.exemption = true;
+    const values = req.form.values.modules.map(m => {
+      return {
+        id: req.model[`module-${m}-id`],
+        modules: [{ module: m, species: [] }],
```
Comment (wheelsandcogs) [inline pages/pil/modules-exempt/index.js]: just call the mapped param `module` :D

File: pages/pil/modules-exempt/schema/index.js [hunk 4]
```diff
@@ -1,10 +1,32 @@
 const { moduleCodes } = require('@asl/constants');
 const { toArray } = require('../../../../lib/utils');
+const content = require('../content');
+
+const options = moduleCodes.map(m => {
```
Comment (wheelsandcogs) [inline pages/pil/modules-exempt/schema/index.js]: `m` -> `moduleName` ?

File: pages/common/assets/sass/style.scss (hunk 1 above)
Comment (wheelsandcogs) [reply pages/common/assets/sass/style.scss]: Yeah I was thinking that, but @lennym had a good reason to keep them separate, but now I can't remember what it was!

File: pages/pil/application/views/exemption-data.jsx (hunk 2 above)
Comment (joefitter) [reply pages/pil/application/views/exemption-data.jsx]: yeah

**Pull request merged**